addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -6618,6 +6618,38 @@ registry.ImageTools = ImageHandlerOption.extend({
     // Options
```

Comment:
@sobo-odoo 
_When applying a shape that should not be squared on an image that was already cropped (2:3 for example), the image should keep its original ratio (so (2:3), and not be reset to 0:0 (as it is the case in your PR) 👍_

as you said in point three, if user applied shape with data-unstretch=true for the first time and wants to apply shape without data=unstretch=true, it will result in squaring of that shapes too. since image knows its current ratio as 1:1.

is this behaviour okay? 
IMO, it should not be like that. Best way is by default shapes with data-unstretch=true comes with 1:1 & without data-unstretch=true comes with 0:0. anyways it's up to specs designer but do consider above behaviour.

Reply:
@rahg-odoo For me this behaviour is ok :wink: The goal of your PR is only to make the shapes that look better in 1:1 ratio always have this ratio when applying them (for example, the circle shape should always be a circle when applied, because an oval is weird looking) => That's why only a subset of the shapes have `data-unstretch=true`, because they need 1:1 to be beautiful :slightly_smiling_face: 
For the other shapes, they always look good, no matter the ratio of the image. The behavior should therefore be like the current one (in general on runbot), which is to keep the ratio of the image when applying these shapes => so 1:1 should stay 1:1, the same way as 2:3 should stay 2:3  :wink: 

An other point about that: imagine a user uploading an image and then cropping it to be 1:1 (because he wants it square). He then wants to add an animated shape (so not a 1:1 shape) => he wanted his image squared, so if it goes back to 0:0, he would have cropped his image for nothing which is not good and confusing :slightly_smiling_face: If he wanted it to be 0:0, he never would have cropped it in the first place :slightly_smiling_face: